system-addon/content-src/components/TopSites/TopSites.jsx

```diff
@@ -478,13 +478,16 @@ class TopSiteForm extends React.Component {
     }
   }
   validateUrl() {
-    const pattern = new RegExp("^(https?:\\/\\/)?" + // protocol
-      "((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|" + // domain name
-      "((\\d{1,3}\\.){3}\\d{1,3}))" + // OR ip (v4) address
-      "(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*" + // port and path
-      "(\\?[;&a-z\\d%_.~+=-]*)?" + // query string
-      "(\\#[\\/-a-z\\d_]*)?$", "i"); // fragment locater
-    return pattern.test(this.state.url);
+    let url = this.state.url;
+    if (url && !url.startsWith("http")) {
```

Comment:
Some hostname could start with "http". I do see this same logic in `cleanUrl` too.. so can you fix that to just check for "http:" or "https:" and then do something like:

```js
validateUrl() {
  try {
    return !!new URL(this.cleanUrl());
  } catch (e) {
    return false;
  }
}
```